docs/v3/codecs/sharding-indexed/v1.0.rst

```diff
@@ -0,0 +1,224 @@
+.. _sharding-indexed-codec-v1:
+
+==========================================
+Sharding codec (version 1.0)
+==========================================
+-----------------------------
+ Editor's draft 18 02 2022
+-----------------------------
+
+Specification URI:
+    https://zarr-specs.readthedocs.io/en/latest/v3/codecs/sharding-indexed/v1.0.html
+
+Issue tracking:
+    `GitHub issues <https://github.com/zarr-developers/zarr-specs/labels/sharding-indexed-codec-v1.0>`-
+
+Suggest an edit for this spec:
+    `GitHub editor <https://github.com/zarr-developers/zarr-specs/blob/main/docs/codecs/sharding-indexed/v1.0.rst>`_
+
+Copyright 2022-Present `Zarr core development team
+<https://github.com/orgs/zarr-developers/teams/core-devs>`_. This work
+is licensed under a `Creative Commons Attribution 3.0 Unported License
+<https://creativecommons.org/licenses/by/3.0/>`_.
+
+----
+
+
+Abstract
+========
+
+This specification defines an implementation of the Zarr codec specification 
+for sharding.
+
+Sharding logically splits chunks ("shards") into sub-chunks ("inner chunks") 
+that can be individually compressed and accessed. This allows to colocate 
+multiple chunks within one storage object, bundling them in shards.
+
+
+Motivation
+==========
+
+In many cases, it becomes inefficient or impractical to store a large number of
+chunks in single files or objects due to the design constraints of the
+underlying storage. For example, the file block size and maximum inode number
+restrict the usage of numerous small files for typical file systems, also cloud
+storage such as S3, GCS, and various distributed filesystems do not efficiently
+handle large numbers of small files or objects.
+
+Increasing the chunk size works only up to a certain point, as chunk sizes need
+to be small for read and write efficiency requirements, for example to stream
```

Comment:
since this solution does not address the write efficiency issue of large chunks, perhaps leave that out